apps/docs/content/guides/auth/server-side/sveltekit.mdx

```diff
@@ -0,0 +1,636 @@
+
+import type { LayoutLoad } from './$types'
+
+export const load: LayoutLoad = async ({ data, depends, fetch }) => {
+  /**
+   * Declare a dependency so the layout can be invalidated, for example, on
+   * session refresh.
+   */
+  depends('supabase:auth')
+
+  const supabase = isBrowser()
+    ? createBrowserClient(PUBLIC_SUPABASE_URL, PUBLIC_SUPABASE_ANON_KEY, {
+        global: {
+          fetch,
+        },
+        cookies: {
+          get(key) {
+            const cookie = parse(document.cookie)
+            return cookie[key]
+          },
+        },
+      })
+    : createServerClient(PUBLIC_SUPABASE_URL, PUBLIC_SUPABASE_ANON_KEY, {
+        global: {
+          fetch,
+        },
+        cookies: {
+          get() {
+            return JSON.stringify(data.session)
+          },
+        },
+      })
+
+  /**
+   * It's fine to use `getSession` here, because on the client, `getSession` is
+   * safe, and on the server, it reads `session` from the `LayoutData`, which
+   * safely checked the session using `safeGetSession`.
+   */
+  const {
+    data: { session },
+  } = await supabase.auth.getSession()
+
+  const {
+    data: { user },
+  } = await supabase.auth.getUser()
+
+  return { session, supabase, user }
+}
+```
+
+```ts src/routes/+layout.server.ts
+import type { LayoutServerLoad } from './$types'
+
+export const load: LayoutServerLoad = async ({ locals: { session } }) => {
+  return {
+    session,
+  }
+}
+```
+
+</CH.Code>
+
+</StepHikeCompact.Code>
+
+</StepHikeCompact.Step>
+
+<StepHikeCompact.Step step={6}>
+
+<StepHikeCompact.Details title="Listen to Auth events">
+
+Set up a listener for Auth events on the client, to handle session refreshes and signouts.
+
+</StepHikeCompact.Details>
+
+<StepHikeCompact.Code>
+
+```svelte src/routes/+layout.svelte
+<script>
+	import { goto, invalidate } from '$app/navigation';
+	import { onMount } from 'svelte';
+
+	export let data;
+	$: ({ session, supabase } = data);
+
+	onMount(() => {
+		const { data } = supabase.auth.onAuthStateChange((_, newSession) => {
+			if (!newSession) {
+				/**
+				 * Queue this as a task so the navigation won't prevent the
+				 * triggering function from completing
+				 */
+				setTimeout(() => {
+					goto('/', { invalidateAll: true });
+				});
+			}
+			if (newSession?.expires_at !== session?.expires_at) {
+				invalidate('supabase:auth');
+			}
+		});
+
+		return () => data.subscription.unsubscribe();
+	});
+</script>
+
+<slot />
+```
+
+</StepHikeCompact.Code>
+
+</StepHikeCompact.Step>
+
+<StepHikeCompact.Step step={7}>
+
+<StepHikeCompact.Details title="Create your first page">
+
+Create your first page. This example page calls Supabase from the server to get a list of countries from the database.
+
+This is an example of a public page that uses publicly readable data.
+
+To populate your database, run the [countries quickstart](/dashboard/project/_/sql/3ab7f123-675a-4342-a5d3-d342ed115a4b) from your dashboard.
```

Comment:
/nit might want to check if this path is fixed - id seems to be `13f0cb1b-1649-4897-aa28-eb1193e792c1` for me
![CleanShot 2024-04-17 at 01 36 21@2x](https://github.com/supabase/supabase/assets/8011761/2f2146ca-087c-42b0-9955-441fafe0f139)

Reply:
damn I was hoping this is fixed, lemme see if I can figure out how to get a permanent link to it...